packages/flutter_tts/tizen/src/flutter_tts_tizen_plugin.cc

```diff
@@ -241,15 +247,18 @@ class FlutterTtsTizenPlugin : public flutter::Plugin {
     std::unique_ptr<flutter::EncodableValue> args =
         std::make_unique<flutter::EncodableValue>(true);
     if (current == TTS_STATE_PLAYING) {
+      plugin->speaking_ = true;
       if (previous == TTS_STATE_READY) {
         plugin->channel_->InvokeMethod("speak.onStart", std::move(args));
       } else if (previous == TTS_STATE_PAUSED) {
         plugin->channel_->InvokeMethod("speak.onContinue", std::move(args));
       }
     } else if (current == TTS_STATE_PAUSED) {
+      plugin->EndSpeaking();
       plugin->channel_->InvokeMethod("speak.onPause", std::move(args));
     } else if (current == TTS_STATE_READY) {
       if (previous == TTS_STATE_PLAYING || previous == TTS_STATE_PAUSED) {
+        plugin->EndSpeaking();
         plugin->channel_->InvokeMethod("speak.onCancel", std::move(args));
```

Comment:
Known issue: The "speak.onCancel" method is always called even when the utterance completes successfully.